restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -562,24 +585,36 @@ public ReactiveDocumentService(
    */
   public Maybe<DocumentResponseWrapper<? extends JsonNode>> executeBuiltInFunction(
       DocumentDB db,
-      String keyspace,
+      String namespace,
       String collection,
       String id,
       ExecuteBuiltInFunction funcPayload,
-      List<PathSegment> path,
+      List<String> path,
       ExecutionContext context) {
     return Maybe.defer(
         () -> {
-          List<String> pathString =
-              path.stream().map(seg -> seg.getPath()).collect(Collectors.toList());
-          Maybe<JsonNode> result = null;
+          // process path
+          List<String> processedPath = processSubDocumentPath(path);
+
+          // note that currently all built in-functions require auth for modify and delete
+          // please adapt if needed in future
           BuiltInApiFunction function = BuiltInApiFunction.fromName(funcPayload.getOperation());
+          authorizeWrite(db, namespace, collection, Scope.MODIFY, Scope.DELETE);
+
+          // based on type switch the result
+          Maybe<JsonNode> result = null;
           if (function == BuiltInApiFunction.ARRAY_PUSH) {
-            result =
-                handlePush(
-                    db, keyspace, collection, id, funcPayload.getValue(), pathString, context);
+            // explicit null check to ensure we have a value to push
+            if (null == funcPayload.getValue()) {
+              throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_INVALID_JSON_VALUE);
+            }
+
+            // TODO Eric what's the desired object here and how would this me mapped anyway?
```

Comment:
Yea I think it is reasonable to make it a JsonNode higher up, since virtually all the time this piece of data has to be JSON of some sort

Reply:
See comment https://github.com/stargate/stargate/pull/1562#discussion_r788530425